### DISCUSSION TITLE
FEATURE REQUEST: Enable loading routes from custom routes file using the group() method in Route facade

### DISCUSSION THREAD

**Discussion by iraklisg** [laravel/framework#34511] [General]
- Laravel Version: 8.4.0
- PHP Version: 7.4.10
- Database Driver & Version: N/A

### Description:
We can load external route files using the `group` method declared in `Illuminate\Routing\RouteRegistar` as follows:

```php
Route::name('myname')->group(__DIR__.'/web/my-routes.php');

``` 

However we cannot load the external route files using the `group` method declared in `Illuminate\Support\Facades\Route` facade:

```php
Route::group(__DIR__.'/web/my-routes.php'); 

// =>  Argument 1 passed to Illuminate\Routing\Router::group() must be of the type array, string given, called in /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php on line 261

```

The problem is that `Route::group()` definition requires to pass `$attributes` as its first argument

```php
Route::group(Closure|string|array $attributes, Closure|string $routes)
```

Would be possible to allow users to load their external route files without being required to pass an empty array of attributes? This would allow users to manage their route prefixes, namespaces, middlewares and names explicitly in an expressive way. For instance: 

```php
// routes/web.php

Route::group(__DIR__ . '/web/my-routes.php');

// routes/web/my-routes.php
Route::get('routeA', [ControllerA::class, 'methodA'])->middleware('middlewareA');
Route::get('routeB', [ControllerA::class, 'methodB'])->name('nameB');
...
```

**Comment by driesvints** [comment]
Heya, thanks for submitting this.

This seems like a feature request or an improvement so I'm moving this to discussions instead. It's best to post these under the discussions tab in the future to get support for your idea. After that you may send a PR to the framework. Please only use this issue tracker to report bugs and issues with the framework.

Thanks!
